app/src/main/java/org/p2p/wallet/striga/finish/StrigaSignupFinishPresenter.kt

```diff
@@ -0,0 +1,21 @@
+package org.p2p.wallet.striga.finish
+
+import kotlinx.coroutines.launch
+import org.p2p.wallet.common.di.AppScope
+import org.p2p.wallet.common.mvp.BasePresenter
+import org.p2p.wallet.common.mvp.MvpView
+import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
+import org.p2p.wallet.striga.user.interactor.StrigaUserInteractor
+
+class StrigaSignupFinishPresenter(
+    private val strigaUserInteractor: StrigaUserInteractor,
+    appScope: AppScope,
+    dispatchers: CoroutineDispatchers,
+) : BasePresenter<MvpView>(dispatchers.ui), StrigaSignupFinishContract.Presenter {
+
+    init {
+        appScope.launch(dispatchers.io) {
+            strigaUserInteractor.loadAndSaveUserStatusData()
```

Comment:
maybe lifecycleScope.launch would be ok?

Reply:
it's safer to launch in a global context, who knows how quickly the response will arrive